lib/performance_platform/gateway/completion_rate.rb

```diff
@@ -25,15 +26,36 @@ def repository
   end
 
   def sms_registered
-    repository.self_sign.with_sms.week_before(date)
+    case @period
+    when "day"
+      repository.self_sign.with_sms.day_before(date)
+    when "week"
+      repository.self_sign.with_sms.week_before(date)
+    when "month"
+      repository.self_sign.with_sms.month_before(date)
+    end
   end
 
   def email_registered
-    repository.self_sign.with_email.week_before(date)
+    case @period
+    when "day"
+      repository.self_sign.with_email.day_before(date)
+    when "week"
+      repository.self_sign.with_email.week_before(date)
+    when "month"
+      repository.self_sign.with_email.month_before(date)
+    end
   end
 
   def sponsor_registered
-    repository.sponsored.week_before(date)
+    case @period
+    when "day"
+      repository.sponsored.day_before(date)
+    when "week"
+      repository.sponsored.week_before(date)
+    when "month"
+      repository.sponsored.month_before(date)
+    end
```

Comment:
I would advise to reduce all these cases with the dynamic method calling suggested above.